src/components/data-table/DataTable.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { useState, ReactNode } from "react";
+import _ from "lodash";
+import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
+import Table from "@material-ui/core/Table";
+
+import Toolbar from "@material-ui/core/Toolbar";
+import Paper from "@material-ui/core/Paper";
+
+import { DataTableHeader } from "./DataTableHeader";
+import { ContextualMenu } from "./ContextualMenu";
+import {
+    TableObject,
+    TablePagination,
+    TableSorting,
+    TableColumn,
+    TableAction,
+    TableNotification,
+} from "./types";
+import { DataTablePagination } from "./DataTablePagination";
+import { DataTableBody } from "./DataTableBody";
+import { sortObjects } from "./utils/sorting";
+import { parseActions, getActionRows, getSelectionMessages } from "./utils/selection";
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        root: {
+            width: "100%",
+        },
+        toolbar: {
+            paddingLeft: 0,
+            paddingRight: 0,
+        },
+        paper: {
+            flex: "1 1 0%",
+            display: "flex",
+        },
+        tableWrapper: {
+            display: "flex",
+            flex: "1 1 0%",
+            overflowX: "auto",
+            paddingBottom: theme.spacing(2),
+        },
+        table: {
+            width: "100%",
+            minWidth: 750,
+        },
+        tablePagination: {
+            flex: "1 1 auto",
+        },
+    })
+);
+
+export interface DataTableProps {
```

Comment:
Looks ok, but we will discuss it, the most important part of the PR is how we interface the public components.